src/spikeinterface/core/segmentutils.py

```diff
@@ -157,7 +157,7 @@ def __init__(self, parent_segments, sampling_frequency, ignore_times=True):
         self.parent_segments = parent_segments
         self.all_length = [rec_seg.get_num_samples() for rec_seg in self.parent_segments]
         self.cumsum_length = [0] + [sum(self.all_length[: i + 1]) for i in range(len(self.all_length))]
-        self.total_length = int(np.sum(self.all_length))
+        self.total_length = int(np.sum(self.all_length, dtype=np.int64))
```

Comment:
The issue with this as a fix would be that if we are doing 10 day recordings in the future then this still might fail. Happy to hear what @h-mayorquin things, but something like:

```
int(sum(list(all_length)))
```
provides us the protection of python integers which shouldn't overflow. We get a performance hit since we won't be working at the numpy vectorized speed, but I think overflows are worse. And overflow behavior is different between OSs < 2.0 (I forget the issue I found that Heberto and I discussed). Hopefully Heberto has a more performant solution than mine though.

Reply:
That's definitely a better solution for robustness!